testing/tests/benchmark/test_testing.py

```diff
@@ -0,0 +1,142 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Benchmark tests for ops-scenario."""
+
+import dataclasses
+import pathlib
+import sys
+
+import ops
+from ops import testing
+
+sys.path.append(
+    str(
+        pathlib.Path(__file__).parent.parent.parent.parent
+        / "test"
+        / "charms"
+        / "test_benchmark"
+        / "src"
+    )
+)
+
+from bcharm import BenchmarkCharm
+
+
+# Note: the 'benchmark' argument here is a fixture that pytest-benchmark
+# automatically makes available to all tests.
+def test_context_explicit_meta(benchmark):
+    ctx = benchmark(testing.Context, ops.CharmBase, meta={"name": "foo"})
+    assert isinstance(ctx, testing.Context)
+
+
+def test_run_no_observer(benchmark):
+    ctx = testing.Context(BenchmarkCharm)
+    benchmark(ctx.run, ctx.on.start(), testing.State())
+    assert len({e.handle.kind for e in ctx.emitted_events}) == 1
+
+
+def test_run_observed(benchmark):
+    ctx = testing.Context(BenchmarkCharm)
+    benchmark(ctx.run, ctx.on.stop(), testing.State())
+    assert len({e.handle.kind for e in ctx.emitted_events}) == 1
+
+
+def test_context_explicit_meta_config_actions(benchmark):
+    ctx = benchmark(
+        testing.Context,
+        ops.CharmBase,
+        meta={"name": "foo"},
+        actions={"act": {"description": "foo"}},
+        config={"options": {"conf": {"type": "int", "description": "bar"}}},
+    )
+    ctx.run(ctx.on.action("act"), testing.State(config={"conf": 10}))
+    assert len({e.handle.kind for e in ctx.emitted_events}) == 1
+
+
+def test_context_autoload_meta(benchmark):
+    ctx = benchmark(testing.Context, BenchmarkCharm)
+    assert isinstance(ctx, testing.Context)
+
+
+def test_many_tests_explicit_meta(benchmark):
+    def mock_pytest():
+        """Simulate running multiple tests against the same charm."""
+        for event in ("install", "start", "stop", "remove"):
+            for _ in range(5):
+                ctx = testing.Context(ops.CharmBase, meta={"name": "foo"})
+                ctx.run(getattr(ctx.on, event)(), testing.State())
+                assert len({e.handle.kind for e in ctx.emitted_events}) == 1
+
+    benchmark(mock_pytest)
+
+
+def test_many_tests_autoload_meta(benchmark):
+    def mock_pytest():
+        """Simulate running multiple tests against the same charm."""
+        for event in ("install", "start", "stop", "remove"):
+            for _ in range(5):
+                ctx = testing.Context(BenchmarkCharm)
+                ctx.run(getattr(ctx.on, event)(), testing.State())
+                assert len({e.handle.kind for e in ctx.emitted_events}) == 1
+
+    benchmark(mock_pytest)
+
+
+def test_lots_of_logs(benchmark):
+    ctx = testing.Context(BenchmarkCharm)
+    benchmark(ctx.run, ctx.on.update_status(), testing.State())
+    assert len(ctx.juju_log) > 200
+
+
+def ditest_full_state(benchmark):
```

Comment:
I imagine this means "disabled test".

Should it be removed before merge?

Reply:
Ah, definitely not meant to be left like this. Thanks - will fix.